src/overlays/actors/ovl_Bg_Mori_Kaitenkabe/z_bg_mori_kaitenkabe.c

```diff
@@ -20,23 +35,142 @@ const ActorInit Bg_Mori_Kaitenkabe_InitVars = {
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 1000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_CONTINUE),
+    ICHAIN_VEC3F_DIV1000(scale, 1000, ICHAIN_STOP),
+};
+
+void BgMoriKaitenkabe_CrossProd(Vec3f* arg0, Vec3f* arg1, Vec3f* arg2) {
+    arg0->x = (arg1->y * arg2->z) - (arg1->z * arg2->y);
+    arg0->y = (arg1->z * arg2->x) - (arg1->x * arg2->z);
+    arg0->z = (arg1->x * arg2->y) - (arg1->y * arg2->x);
+}
+
+void BgMoriKaitenkabe_Init(Actor *thisx, GlobalContext *globalCtx) {
+    s32 pad;
+    BgMoriKaitenkabe* this = THIS;
+    s32 localConst = 0;
+    
+    //Forest Temple object 【Rotating Wall (arg_data: 0x% 04x)】 appears
+    osSyncPrintf("◯◯◯森の神殿オブジェクト【回転壁(arg_data : 0x%04x)】出現 \n", this->dyna.actor.params);
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyInfo_SetActorMove(&this->dyna, 0);
+    DynaPolyInfo_Alloc(&D_060063B8, &localConst);
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, localConst);
+    this->moriTexObjIndex = Object_GetIndex(&globalCtx->objectCtx, OBJECT_MORI_TEX);
+    if (this->moriTexObjIndex < 0) {
+        Actor_Kill(&this->dyna.actor);
+        // 【Rotating wall】 Bank danger!
+        osSyncPrintf("【回転壁】 バンク危険！(%s %d)\n", "../z_bg_mori_kaitenkabe.c", 176);
+    } else {
+        this->actionFunc = BgMoriKaitenkabe_WaitForMoriTex;
+    }
+}
+
+void BgMoriKaitenkabe_Destroy(Actor* thisx, GlobalContext *globalCtx) {
+    s32 pad;
+    BgMoriKaitenkabe* this = THIS;
+    
+    DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+}
+
+void BgMoriKaitenkabe_WaitForMoriTex(BgMoriKaitenkabe *this, GlobalContext *globalCtx) {
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->moriTexObjIndex)) {
+        BgMoriKaitenkabe_SetupWait(this);
+        this->dyna.actor.draw = BgMoriKaitenkabe_Draw;
+    }
+}
+
+void BgMoriKaitenkabe_SetupWait(BgMoriKaitenkabe* this) {
+    this->actionFunc = BgMoriKaitenkabe_Wait;
+    this->timer = 0;
+}
+
+void BgMoriKaitenkabe_Wait(BgMoriKaitenkabe *this, GlobalContext *globalCtx) {
+    Vec3f sp3C;
+    Vec3f sp30;
+    Vec3f sp24;
+    Player* player = PLAYER;
+
+    if (0.001f < this->dyna.unk_150) {
+        this->timer++;
+        if ((this->timer > 28) && !Player_InCsMode(globalCtx)) {
+            BgMoriKaitenkabe_SetupRotate(this);
+            func_8002DF54(globalCtx, &this->dyna.actor, 8);
+            Math_Vec3f_Copy(&this->playerPos, &player->actor.posRot.pos);
+            sp3C.x = Math_Sins(this->dyna.unk_158);
+            sp3C.y = 0.0f;
+            sp3C.z = Math_Coss(this->dyna.unk_158);
+            sp30.x = this->dyna.actor.posRot.pos.x - player->actor.posRot.pos.x;
+            sp30.y = 0.0f;
+            sp30.z = this->dyna.actor.posRot.pos.z - player->actor.posRot.pos.z;
+            BgMoriKaitenkabe_CrossProd(&sp24, &sp3C, &sp30);
```

Comment:
`this->dyna.unk_158` is set by (at least) [func_8002DFA4](https://github.com/zeldaret/oot/blob/14191307e96cc915fa5a67704eba16421cf4deb2/src/code/z_actor.c#L1059) called by (at least) [func_8084B840](https://github.com/zeldaret/oot/blob/14191307e96cc915fa5a67704eba16421cf4deb2/src/overlays/actors/ovl_player_actor/z_player.c#L10793)
so it seems to be the `PLAYER->actor.posRot.rot.y` value last time Link did something wall-related (grab, I would wildly guess?)

(Not sure the following, based on guesses, should be put into the code)

So `sp3C` would be `pushDirection`
Actor origin of kaitenkabe is the center of the room at floor level, `sp30` is `oppositePushPosition` (pushPosition would be relative to actor origin, idk how to name its opposite)
then `sp24 = sp3C × sp30 = pushDirection × oppositePushPosition = pushDirection × -pushPosition = pushPosition × pushDirection` indeed has its y component positive when the room should rotate around +y and negative in the other case

oh and the same functions I highlighted also seem to hint that `unk_150` could be the pressure applied (going up when link keeps pushing)

Reply:
Yeah, this is essentially a torque calculation, and the y-component of torque corresponds to rotations around the y-axis.